routes/artifacts.js

```diff
@@ -190,6 +190,18 @@ api.declare({
         details.bucket  = this.privateBucket.bucket;
       }
       details.prefix    = [taskId, runId, name].join('/');
+      if (input.contentMD5 || input.contentSHA256) {
```

Comment:
perhaps both of these should be defined as optional headers in the description above and then we can change it to "*must*" when it's required.